docs/papers/architecture-whitepaper.md

```diff
@@ -0,0 +1,1709 @@
+    1. The developers and operators of digital wallets must not be able to observe or track transactions undertaken by wallet owners.
+1. **Privacy by Design**: Digital wallets should support privacy preserving mechanisms.
+1. **Minimal Disclosure:** Digital wallets must disclose the minimal amount of data needed by the use case.
+
+### Interoperable
+
+1. **Multi-protocol Support**: The wallet must work across different protocols, ensuring that it can handle interactions with various blockchain platforms, identity verification systems, and payment networks.
+1. **Network Neutrality**: Operate across various blockchain networks and ledger types, including permissioned and permissionless systems, to support a wide array of applications.
+1. **Extensive Compatibility**: It should be compatible with a wide range of devices and platforms, ensuring users can access any chosen interface that supports the needed functionality.
+1. **Multi-framework / Standard Support**: It should provide a range of identity solutions, supporting various levels of verification and credential types to cater to different requirements of services and jurisdictions.
+1. **Protocol Agnosticism**: The wallet must be capable of interacting with various protocols and standards without favoring one over another, ensuring broad compatibility.
+1. **Legacy System Integration**: Provide bridges or adapters to allow for interaction with legacy systems, ensuring that users are not excluded from the wallet's ecosystem.
+1. **Automated Compatibility Checks**: Implement mechanisms to automatically verify the compatibility of external systems and services, facilitating smooth integrations and updates.
+1. **Standardized Data Formats**: Adopt standardized data formats for storing and exchanging information to simplify the process of sharing data across different systems.
+
+### Portable
+
+1. **Cloud and Local Storage**: The wallet should support both cloud-based and local storage options, giving users the flexibility to choose where their data resides.
+1. **Data Export and Import Capabilities**: Users should be able to easily export their data from the wallet and import it into another wallet or system, using standard, non-proprietary formats.
+1. **Backup and Recovery Options**: Provide secure and intuitive backup and recovery options to ensure that users can regain access to their wallet even if their primary device is lost, stolen, or damaged.
+1. **User-Centric Design**: Maintain a user-centric approach that prioritizes ease of use in transferring the wallet from one environment to another without technical barriers.
+1. **Decoupling From Hardware Constraints**: Design the wallet in such a way that it is not bound to specific hardware features, making it more adaptable to various technological contexts.
```

Comment:
Totally agree with that principle but I fear that it could be misinterpreted that specific hardware feature should not be used. Thinking about TEEs, HSMs and proof of ownership (device bound credentials)

Reply:
Any suggestions for re-wording? It is definitely not the intent to limit the use of TEEs/HSMs, but rather trying to ensure that we have the ability to switch in libraries that support different hardware.

Reply:
Maybe something like "Design the wallet in such a way that it is not bound to specific hardware, making it interoperable with various technological contexts."?

After re-reading it, I think the risk of misinterpreting the sentence is really minimal. Both option work for me.